### ISSUE TITLE
Problem when passing undefined

### ISSUE THREAD

**Issue by henrikra** [yayoc/optional-chain#4]
I got error `Argument of type '"first"' is not assignable to parameter of type 'never'.` with code like this:
![image](https://user-images.githubusercontent.com/12229968/43149123-65512018-8f6f-11e8-8e25-f7d54b0f6a43.png)

If I passed `person` everything was fine but not when I passed `person.name` to `optional` function.

Then I looked at the types inside node_modules which were like this
```typescript
export declare enum OptionType {
    None = 0,
    Some = 1
}
export declare const optional: <T>(value: T) => Option<T>;
export declare type Diff<T, U> = T extends U ? never : T;
export declare type NonNullable<T> = Diff<T, null | undefined>;
export declare class Option<T> {
    readonly value: T;
    readonly type: OptionType;
    constructor(value: T);
    k<K extends keyof T>(key: K): Option<NonNullable<T[K]>>;
    i: T extends Array<infer I> ? (index: number) => Option<I> : Option<undefined>;
    get(): T;
    match({ some, none }: {
        some: (v: T) => any;
        none: (v: T) => any;
    }): any;
    getOrElse(value: T): T;
}
```

I changed this line
```typescript
export declare const optional: <T>(value: T) => Option<T>;
```
 
to this

```

export declare const optional: <T>(value: T) => Option<NonNullable<T>>;
```

then it worked

**Comment by henrikra** [comment]
@yayoc Do you understand what I am saying here? :D

**Comment by henrikra** [comment]
@yayoc Are you still on holiday? :D

**Comment by yayoc** [comment]
@henrikra Sorry, for the late reply.
Yeah, you are right, I am facing the same issue. Let me update the package as you said.
Thanks

**Comment by yayoc** [comment]
Published [v0.1.2](https://github.com/yayoc/optional-chain/releases/tag/v0.1.2) :D

**Comment by henrikra** [comment]
Good to hear you are back!
